src/DeviceInterface/WindowsInterface.js

```diff
@@ -0,0 +1,98 @@
+import InterfacePrototype from './InterfacePrototype.js'
+import {OverlayUIController} from '../UI/controllers/OverlayUIController.js'
+import {CloseAutofillParentCall, GetAutofillDataCall} from '../deviceApiCalls/__generated__/deviceApiCalls.js'
+
+/**
+ * @typedef {import('../deviceApiCalls/__generated__/validators-ts').GetAutofillDataRequest} GetAutofillDataRequest
+ */
+
+export class WindowsInterface extends InterfacePrototype {
+    ready = false;
+    /** @type {AbortController|null} */
+    _abortController = null;
+    /**
+     * @deprecated This is too early, and will be removed eventually.
+     * @returns {Promise<boolean>}
+     */
+    async isEnabled () {
+        return true
+    }
+
+    isEnabledViaSettings () {
+        return Boolean(this.settings.enabled)
+    }
+
+    postInit () {
+        const cleanup = this.scanner.init()
+        this.addLogoutListener(cleanup)
+        this.ready = true
+    }
+
+    createUIController () {
+        /**
+         * If we get here, we're just a controller for an overlay
+         */
+        return new OverlayUIController({
+            remove: async () => this._closeAutofillParent(),
+            show: async (details) => this._show(details)
+        })
+    }
+
+    /**
+     * @param {GetAutofillDataRequest} details
+     */
+    async _show (details) {
+        const { mainType } = details
+        // prevent overlapping listeners
+        if (this._abortController && !this._abortController.signal.aborted) {
+            this._abortController.abort()
+        }
+        this._abortController = new AbortController()
+        this.deviceApi.request(new GetAutofillDataCall(details), { signal: this._abortController.signal })
```

Comment:
Nice. Is this ever needed on other platforms or was it something specific to Windows?

Reply:
This was specific to windows - because of the way their messaging works, you can run into a situation where you have multiple event listeners firing when tabbing through form fields (for example).